policies.go

```diff
@@ -146,6 +146,9 @@ const (
 // users to implement their own logic to determine if a query can be attempted
 // again.
 //
+// RetryPolicy is a work in progress. Notably, query idempotence currently has
+// no effect on retry policies.
+//
```

Comment:
I came here exactly for this reason because I've been reviewing #1178 and I knew that there was something about query idempotency not being taken into account. Just know that I've read your comment here and will refer to it in the review.